lib/rules/max-statements.js

```diff
@@ -84,9 +84,18 @@ module.exports = {
          */
         function reportIfTooManyStatements(node, count, max) {
             if (count > max) {
+                const messageEnd = " has too many statements ({{count}}). Maximum allowed is {{max}}.";
+                let name = "This function";
+
+                if (node.id) {
+                    name = `Function '${node.id.name}'`;
+                } else if (node.parent.type === "MethodDefinition" || node.parent.type === "Property") {
+                    name = `Function '${node.parent.key.name}'`;
+                }
```

Comment:
Lines 88-94 seem like they might be reused across other rules where we need to extract a function name. The logic in those lines should be in ast-utils at some point. Sorry for being unclear.